app/domain/authentication/common/extract_resource_restrictions.rb

```diff
@@ -0,0 +1,106 @@
+require 'command_class'
+
+module Authentication
+  module Common
+
+    ExtractResourceRestrictions = CommandClass.new(
+      dependencies: {
+        resource_restrictions_class: ResourceRestrictions,
+        role_class:                  ::Role,
+        resource_class:              ::Resource,
+        logger:                      Rails.logger
+      },
+      inputs:   %i(authenticator_name service_id host_name account)
+    ) do
+
+      def call
+        @logger.debug(LogMessages::Authentication::ExtractingRestrictionsFromResource.new(
+          @authenticator_name,
+          @host_name
+        ))
+
+        annotations = fetch_role_annotations
+        resource_restrictions_hash = extract_resource_restrictions_from_annotations(annotations)
+        resource_restrictions = create_resource_restrictions_object(resource_restrictions_hash)
+
+        @logger.debug(LogMessages::Authentication::ExtractedResourceRestrictions.new(resource_restrictions.names))
+
+        resource_restrictions
+      end
+
+      private
+
+      def fetch_role_annotations
```

Comment:
this should be `def role_annotations`. or is it `fetch` to indicate that we have a critical action here of going to the db?

Reply:
the latter, yes.